### PR TITLE
[BEAM-6909] Add location support for BigQueryWrapper._get_query_resul…

### PR DESCRIPTION
**Pull request by ryanyuan**
When a BQ Job is created outside US or EU, the _get_query_results() in bigquery_tools.py always return 404.

This is to patch the _get_query_results() so that it can support location as parameter.

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by ryanyuan** [comment]
@pabloem

**Comment by pabloem** [comment]
Hi @ryanyuan : ) I'm very glad to have your contribution. I suspect you may have filed a support ticket about this? I'll take a quick look at your PR and see what may be failing the tests.

**Comment by pabloem** [comment]
It's only a few lint issues. https://scans.gradle.com/s/pu5rhlrpchesa/console-log?task=:beam-sdks-python:lintPy27#L26

would you consider fixing them and pushing the change?

File: sdks/python/apache_beam/io/gcp/bigquery_tools.py [hunk 1]
```diff
@@ -347,16 +347,21 @@ def _start_query_job(self, project_id, query, use_legacy_sql, flatten_results,
...
-        maxResults=max_results)
+                         page_token=None, max_results=10000, location=None):
+    if location:
+      request = bigquery.BigqueryJobsGetQueryResultsRequest(
+          jobId=job_id, pageToken=page_token, projectId=project_id,
+          maxResults=max_results, location=location)
+    else:
+      request = bigquery.BigqueryJobsGetQueryResultsRequest(
+          jobId=job_id, pageToken=page_token, projectId=project_id,
+          maxResults=max_results)
```
Comment (pabloem) [inline sdks/python/apache_beam/io/gcp/bigquery_tools.py]: I believe it's okay to do location=None as well, so you don't need to have the `if`/`else` path here.

**Comment by ryanyuan** [comment]
Run Portable_Python PreCommit

**Comment by ryanyuan** [comment]
Run Python PreCommit

**Comment by ryanyuan** [comment]
Run Python PreCommit

**Comment by ryanyuan** [comment]
Run Python_PVR_Flink PreCommit

**Comment by ryanyuan** [comment]
Run Python PreCommit

**Comment by pabloem** [comment]
Run Python PreCommit

File: sdks/python/apache_beam/io/gcp/bigquery_tools.py (hunk 1 above)
Comment (ryanyuan) [reply sdks/python/apache_beam/io/gcp/bigquery_tools.py]: @pabloem  Yeah you are right. Thanks for the suggestion!

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Thanks Ryan. This change LGTM. I will wait a day or two before merging, because we're cutting a release, and I want to have this change until the next release to test it.

**Comment by ryanyuan** [comment]
@pabloem all good 👍

**Comment by pabloem** [comment]
Run Python postcommit

**Comment by pabloem** [comment]
Oddly, this change is causing trouble in PostCommit tests, so we will need to review what's going on. I'll look into this next week.

**Comment by ryanyuan** [comment]
Run Python postcommit

**Comment by ryanyuan** [comment]
@pabloem I’ve fixed the bug. It was the parameter’s position issue.

**Comment by pabloem** [comment]
Very cool. Really happy to have this.

**Pull request merged**